crates/ipc-router-wip/src/lib.rs

```diff
@@ -119,143 +186,60 @@ impl<U> Router<U> {
-            $($params: serde::de::DeserializeOwned,)*
-            R: serde::Serialize
-        {
-            fn into_func(self) -> Definition<T> {
-                Box::new(move |caller, params, tx| {
-                    log::debug!("Deserializing parameters...");
-                    let ($($params,)*) = postcard::from_bytes(params)?;
-
-                    log::debug!("Calling handler...");
-                    let out = self(caller, $($params),*)?;
-
-                    log::debug!("Serializing response...");
-                    let out = postcard::to_allocvec(&out)?;
-
-                    tx.send(out)?;
-
-                    Ok(())
-                })
-            }
-        }
-    }
-}
-
-macro_rules! for_each_function_signature {
-    ($mac:ident) => {
-        $mac!(0);
-        $mac!(1 A1);
-        $mac!(2 A1 A2);
-        $mac!(3 A1 A2 A3);
-        $mac!(4 A1 A2 A3 A4);
-        $mac!(5 A1 A2 A3 A4 A5);
-        $mac!(6 A1 A2 A3 A4 A5 A6);
-        $mac!(7 A1 A2 A3 A4 A5 A6 A7);
-        $mac!(8 A1 A2 A3 A4 A5 A6 A7 A8);
-        $mac!(9 A1 A2 A3 A4 A5 A6 A7 A8 A9);
-        $mac!(10 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10);
-        $mac!(11 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10 A11);
-        $mac!(12 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10 A11 A12);
-        $mac!(13 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10 A11 A12 A13);
-        $mac!(14 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10 A11 A12 A13 A14);
-        $mac!(15 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10 A11 A12 A13 A14 A15);
-        $mac!(16 A1 A2 A3 A4 A5 A6 A7 A8 A9 A10 A11 A12 A13 A14 A15 A16);
-    };
-}
-
-for_each_function_signature!(impl_into_func);
-
 pub trait BuilderExt {
     #[must_use]
     fn ipc_router<U: Send + Sync + 'static>(self, router: Router<U>) -> Self;
 }
 
-impl<R: Runtime> BuilderExt for tauri::Builder<R> {
+impl<R: tauri::Runtime> BuilderExt for tauri::Builder<R> {
     fn ipc_router<U: Send + Sync + 'static>(self, router: Router<U>) -> Self {
-        self.manage(Mutex::new(router))
-            .register_uri_scheme_protocol("ipc", |app, req| {
-                let res = uri_scheme_handler_inner::<U, _>(app, req);
-
-                log::debug!("call result {:?}", res);
-
-                let mut resp = match res {
-                    Ok(val) => {
-                        let mut resp = tauri::http::Response::new(val);
-                        resp.set_status(tauri::http::status::StatusCode::OK);
-                        resp.set_mimetype(Some("application/octet-stream".to_string()));
-                        resp
-                    }
-                    Err(err) => {
-                        let mut resp = tauri::http::Response::new(err.to_string().into_bytes());
-                        resp.set_status(tauri::http::status::StatusCode::BAD_REQUEST);
-                        resp
-                    }
+        let router = Arc::new(router);
+
+        self.register_asynchronous_uri_scheme_protocol("ipc", move |_app, req, responder| {
```

Comment:
nice you already updated to the new API :)

Reply:
Yep!